server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -211,6 +218,19 @@ public CreateIndexRequest settings(Map<String, ?> source) {
         return this;
     }
 
+    /**
+     * Set the mapping for this index
+     *
+     * The mapping should be in the form of a JSON string, with an outer _doc key
+     * <pre>
+     *     .mapping("{\"_doc\":{\"properties\": ... }}")
+     * </pre>
+     */
+    public CreateIndexRequest mapping(String mapping) {
```

Comment:
One the client side [we are not nesting the mapping definition under the type](https://github.com/elastic/elasticsearch/blob/master/client/rest-high-level/src/main/java/org/elasticsearch/client/indices/CreateIndexRequest.java#L156) . Are we ok with this difference?

Reply:
RestCreateIndexAction deals with adding this nesting in `prepareMappings`: https://github.com/elastic/elasticsearch/blob/37a997a9a960d8464185672ea56c76374991e896/server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestCreateIndexAction.java#L68

Reply:
It is, but I want to do this in a way that makes backports simple - just removing things will break more or less every backport to 7x from master.